score-server/src/main/java/bio/overture/score/server/repository/s3/S3DownloadService.java

```diff
@@ -105,57 +105,88 @@ public ObjectSpecification download(
       // Construct ObjectSpecification for actual object in /data logical folder
       val objectKey = ObjectKeys.getObjectKey(dataDir, objectId);
 
-      List<Part> parts;
-      if (forExternalUse) {
-        // Return as a single part - no matter how large
-        parts = partCalculator.specify(0L, -1L);
-      } else {
-        parts = partCalculator.divide(offset, length);
-      }
       if (objectSpec == null) {
         ObjectMetadata metadata =
             s3Client.getObjectMetadata(
                 bucketNamingService.getObjectBucketName(objectId), objectKey.getKey());
+        // fetch the list of multipartUploads for the bucket
+        ListMultipartUploadsRequest listMultipartUploadsRequest =
+            new ListMultipartUploadsRequest(bucketNamingService.getObjectBucketName(objectId));
+        List<MultipartUpload> multipartUploads =
+            s3Client.listMultipartUploads(listMultipartUploadsRequest).getMultipartUploads();
```

Comment:
It is unclear why generating a new upload request with the s3 client would solve our problem here. At the time we are downloading we need to get information about our object from the information available in the bucket, generating a new request will not allow it to calculate part information about our existing file.

I spent time to read through how the code is expecting to generate this part list from the `.meta` file... they were actually storing a list of uploaded parts in that file, and importantly they were recording the md5 value for each of those parts. Unfortunately we don't have this file so we aren't able to return part information with md5 hashes.

Reading more through this code, we already have a mechanism that generates the list of parts using `SimplePartCalculator.divide`. Our whole issue was that this was being passed a length argument with value 0. Fixing this should solve our whole problem. Our parts won't have md5 hashes but they should exist.